crates/sncast/tests/docs_snippets/validation.rs

```diff
@@ -0,0 +1,75 @@
+use docs::validation::{
+    extract_matches_from_directory, extract_matches_from_file, get_parent_dir,
+    snippet_to_command_args,
+};
+use regex::Regex;
+use tempfile::tempdir;
+
+use crate::helpers::runner::runner;
+
+#[test]
+fn test_docs_snippets() {
+    let tempdir: tempfile::TempDir = tempdir().expect("Unable to create a temporary directory");
+
+    let root_dir_path = get_parent_dir(2);
+    let docs_dir_path = root_dir_path.join("docs/src");
+    let sncast_readme_path = root_dir_path.join("crates/sncast/README.md");
+
+    let re = Regex::new(r"(?ms)```shell\n\$ (sncast .+?)\n```").expect("Invalid regex pattern");
+    let extension = Some("md");
+    let docs_snippets = extract_matches_from_directory(&docs_dir_path, &re, extension)
+        .expect("Failed to extract sncast command snippets");
+
+    let readme_snippets = extract_matches_from_file(&sncast_readme_path, &re)
+        .expect("Failed to extract sncast command snippets");
+
+    let snippets = docs_snippets
+        .into_iter()
+        .chain(readme_snippets)
+        .collect::<Vec<String>>();
+
+    let skipped_args = [
+        // snippet "$ sncast <subcommand>"
+        vec!["<subcommand>"],
+        // snippet with interactive account import example
+        vec![
+            "account",
+            "import",
+            "--url",
+            "http://127.0.0.1:5050",
+            "--name",
+            "account_123",
+            "--address",
+            "0x1",
+            "--type",
+            "oz",
+        ],
+    ];
```

Comment:
I wonder if we should be defining what to skip here. 
This is probably okay as an initial solution but I think ideally snippets to be skipped should probably somehow marked in markdown instead.

Reply:
You can create a follow-up task for that if this cannot be resolved easily.

Reply:
> I wonder if we should be defining what to skip here. This is probably okay as an initial solution but I think ideally snippets to be skipped should probably somehow marked in markdown instead.

Yup, makes sense.